### PR TITLE
Add Zabbix inventory support

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
Add Zabbix inventory handling to zabbix execution module
Support Zabbix inventory handling in zabbix_host state

### Tests written?
No

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/zabbix.py [hunk 1]
```diff
@@ -1058,6 +1058,97 @@ def host_update(hostid, **connection_args):
         return ret
 
 
+def hostinventory_get(hostids, **connection_args):
+    '''
+    Retrieve host inventory according to the given parameters.
+    See: https://www.zabbix.com/documentation/2.4/manual/api/reference/host/object#host_inventory
+
+    .. versionadded:: Oxygen
```
Comment (rallytime) [inline salt/modules/zabbix.py]: This should be updated to `Fluorine` as we've completed the feature freeze for `oxygen` already.

File: salt/modules/zabbix.py [hunk 2]
```diff
@@ -1058,6 +1058,97 @@ def host_update(hostid, **connection_args):
...
+        return ret
+
+
+def hostinventory_set(hostid, **connection_args):
+    '''
+    Update host inventory items
+    NOTE: This function accepts all standard host: keyword argument names for inventory
+    see: https://www.zabbix.com/documentation/2.4/manual/api/reference/host/object#host_inventory
+
+    .. versionadded:: Oxygen
```
Comment (rallytime) [inline salt/modules/zabbix.py]: Same comment here.

File: salt/modules/zabbix.py [hunk 3]
```diff
@@ -1058,6 +1058,97 @@ def host_update(hostid, **connection_args):
         return ret
 
 
+def hostinventory_get(hostids, **connection_args):
```
Comment (rallytime) [inline salt/modules/zabbix.py]: I think `host_inventory_get` might be a better name here. I think it's easier to read.

File: salt/modules/zabbix.py [hunk 4]
```diff
@@ -1058,6 +1058,97 @@ def host_update(hostid, **connection_args):
...
+            params = _params_extend(params, **connection_args)
+            ret = _query(method, params, conn_args['url'], conn_args['auth'])
+            return ret['result'][0]['inventory'] if len(ret['result'][0]['inventory']) > 0 else False
+        else:
+            raise KeyError
+    except KeyError:
+        return ret
+
+
+def hostinventory_set(hostid, **connection_args):
```
Comment (rallytime) [inline salt/modules/zabbix.py]: `host_inventory_set` on this function is easier to read, IMHO, as well.

**Review by rallytime — CHANGES_REQUESTED** [review]
Thanks for this @mchugh19!

I have a couple of small comments.

Can you also write some tests for these new functions?

**Comment by mchugh19** [comment]
I'm having a bit of trouble coming up with the mocking necessary for the _query method. Could anyone guide me, and I'll write up the test cases?

**Comment by mchugh19** [comment]
Took another crack at it and got unit tests going. Added tests for the existing user_* and the newly added host_inventory_* methods.

**Comment by rallytime** [comment]
@mchugh19 There's a lint error here now: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/19090/violations/

Looks like there's also a related test failure: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-cent7-py3/2092/

**Comment by mchugh19** [comment]
Lint output cleaned up. I had an issue running pylint from python2, but just corrected that.

**Comment by mchugh19** [comment]
The referenced test failures seem unrelated as they had to do with other testing scripts like vmware.

**Comment by mchugh19** [comment]
I think all requested updates have been made. Is there anything still outstanding?

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
@mchugh19 This looks good to me but the tests timed out before I could double check them. I'll keep an eye on this so we can get it in. Thanks!

**Pull request merged**